spritesheet-manager.js

```diff
@@ -29,13 +33,35 @@ class SpritesheetManager {
     }
     return spritesheet;
   }
+
   async getSpriteSheetForAppUrlAsync(appUrl, opts) {
-    let spritesheet = this.spritesheetCache.get(appUrl);
-    if (!spritesheet) {
-      spritesheet = await this.getSpriteSheetForAppUrlInternal([appUrl, opts]);
-      this.spritesheetCache.set(appUrl, spritesheet);
+    const spritesheet = this.spritesheetCache.get(appUrl);
+
+    if (spritesheet) return spritesheet;
+
+    return new Promise((resolve) => {

```

Comment:
IMO not the best to have polymorphic functions like this; it's more complicated to read and make strongly typed.

It could just `await` the result and return it at the top level.